Home work4

ViacheslavDobriy opened the pull request.

ViacheslavDobriy committed:
Created class People and added some fields. I successfully created object with class People and used Constructor.

ViacheslavDobriy committed:
Added 5 more people for competition.

ViacheslavDobriy commented:
Создал класс people и создал 6 people объектов, смог получить имена, их возраст и силу на консоли. Есть соответствующие методы для их ввода. Возраст и сила рандомно генерируются. А также есть класс competition где есть точка входа main. Имя вводится в точке входа в программу, при создании объекта.

ViacheslavDobriy committed:
Added class Team. Sorted sportsmens to 2 teams. Got info about team.

ViacheslavDobriy committed:
Added classes Obstacle and ObstacleCourse, set info about it and got info in console.

ViacheslavDobriy committed:
Commented code from seminar 4

ViacheslavDobriy committed:
Merge remote-tracking branch 'origin/HomeWork4' into HomeWork4

ViacheslavDobriy committed:
After doing obstacles peoples in both teams lose power. It means that program is working!!!!
There are 2 teams, 3 people in each and three different obstacles.

ViacheslavDobriy commented:
На данный момент моя программа способна вычесть энергию каждого спортсмена из команды, при прохождении каждого препятствия по отдельности. Программа не дает конкретного ответа какая из команд победила. До конца не понимаю зачем нужен абстрактный класс и как его грамотно использовать в совокупности с обычным классом. У меня есть класс Препятствие и есть класс Полоса препятствий, второй из них является абстрактным. Поэтому создание всех объектов всех классов я организовал в точке входа, просто разным людям давал разные характеристики, разным командам - разные имена и разным препятствиям - разные запросы по силе прямо в main.
Также не знаю правильно ли объявлять объект одного класса в другом классе. Например в моей работе, в классе Team, я объявляю объект класса People[].

ViacheslavDobriy committed:
Program is giving answer, which team won!

ViacheslavDobriy committed:
some changing in project

ViacheslavDobriy commented:
Разобрался немного как использовать абстрактный класс. Есть абстрактный класс Полоса препятствий, и как я понимаю я создал 3 дочерних класса к нему, бег, плавание и прыжки, которые наследуют от него всего методы, и которые требуют реализации всех абстрактных методов. 
Не понимаю, как сделать так, чтобы разные дочерние классы содержали разные характеристики, если взять плавание например, то как в этот класс вставить глубину бассейна, не затрагивая другие дочерние классы.
Для выявление команды, которая победила, программа высчитывает количество игроков, сила которых неотрицательна. В какой команде больше таких игроков, та команда и победила. Если же количество очков одинаковое, то победила дружба.

snow-wh commented on src/main/java/HomeWorks/HomeWork4/Running.java:
```diff
@@ -0,0 +1,38 @@
+import java.util.Scanner;
+
+public class Running extends ObstacleCourse{
+
```
Тут нужно было создать конструктор, который задает какие нибудь параметры сложности 
пример 

```
private int width;
private int length ; // создаем локальные переменные

Running(int width, int length ){
    this.width=width;
    this.length =length ; //дальше используешь там где тебе нужно эти уровни сложности и тд 
}
```

snow-wh commented on src/main/java/HomeWorks/HomeWork4/Competition.java:
```diff
@@ -0,0 +1,65 @@
+
+        ObstacleCourse swimming = new Swimming();
+
+        ObstacleCourse jumping = new Jumping();
```
а абстрактный класс тут нужен для создания полосы препятствий 

создай еще один класс с конструктором 

```
LineCourse (ObstacleCourse ... course){ 
}
```
//Он примет список препятствий 
// и в этом же классе реализуй метод                   
//competied или что нибудь такое который будет принимать два класса Team 
ну и в main будет что то типо 

```
LineCourse  line = new LineCourse(running,swimming,jumping);
line.competied(team1,team2);
sout(line.getResult())
```

ViacheslavDobriy replied:
А можно ли в конструкторе вызывать методы, которые нам необходимы? Я создал новый класс, который принимает три объекта - три препятствия. 
Он собирает три препятствия в один лист. На этом мой конструктор завершает работу. 
А как мне вызвать метод competition из нового мною созданного класса, не упоминая ему ни об одной переменной из этого же класса. 
Получается что у меня конструктор, он какой то бесполезный, я не могу от него передать этот лист в метод competition.

ViacheslavDobriy committed:
Added one more class LineOfCourses, which consists of constructor and method Competition.
1) Figure out how to solve problem about useless constructor LineOfCourses, because i can not call method competition if i dont know list of courses. Now it's solved in method main...

ViacheslavDobriy committed:
Added field List<ObstacleCourse> in LineOfCourses class.
Delete extra value in method Competition of LineOfCourses class.

ViacheslavDobriy committed:
Transferred method WhoIsWinner in class LineOfCourses and called this method in main.
Way to upgrade:
 1) add more courses;
 2) add different line of courses with one classes.
 3) add more conditions for people;
 4) add more requiring for courses;

ViacheslavDobriy committed:
Shifted all tasks from 1 homework to package with the same name.

ViacheslavDobriy committed:
Commit extra changing

ViacheslavDobriy committed:
Merge remote-tracking branch 'origin/main'

ViacheslavDobriy committed:
Commit extra changing
